PyInstaller/cli/opts.py

```diff
@@ -0,0 +1,448 @@
+        'and the child (e.g., via a process group) to avoid signalling the child twice.',
+    )(cmd)
+    cmd = optgroup('Rarely used special options')(cmd)
+
+    # Group: macOS specific options
+
+    cmd = optgroup.option(
+        '--argv-emulation',
+        is_flag=True,
+        default=False,
+        help='Enable argv emulation for macOS app bundles. If enabled, the initial open document/URL '
+        'event is processed by the bootloader and the passed file paths or URLs are appended '
+        'to sys.argv.',
+    )(cmd)
+    cmd = optgroup.option(
+        '--osx-bundle-identifier',
+        'bundle_identifier',
+        type=str,
+        help='Mac OS .app bundle identifier is used as the default unique program name for code signing '
+        'purposes. The usual form is a hierarchical name in reverse DNS notation. For example: '
+        'com.mycompany.department.appname (default: first script\'s basename)'
+    )(cmd)
+    cmd = optgroup.option(
+        '--target-arch',
+        'target_arch',
+        metavar='ARCH',
+        help='Target architecture (macOS only; valid values: x86_64, arm64, universal2). Enables switching '
+        'between universal2 and single-arch version of frozen application (provided python '
+        'installation supports the target architecture). If not target architecture is not specified, '
+        'the current running architecture is targeted.',
+    )(cmd)
+    cmd = optgroup.option(
+        '--codesign-identity',
+        metavar='IDENTITY',
+        help='Code signing identity (macOS only). Use the provided identity to sign collected binaries and '
+        'generated executable. If signing identity is not provided, ad-hoc signing is performed instead.',
+    )(cmd)
+    cmd = optgroup.option(
+        '--osx-entitlements-files',
+        'entitlements_file',
+        metavar='FILE',
+        help='Entitlements file to use when code-signing the collected binaries (macOS only).',
+    )(cmd)
+    cmd = optgroup('macOS specific options')(cmd)
+
+    # Group: Windows side-by-side assembly searching options
+
+    cmd = optgroup.option(
+        '--win-private-assemblies',
+        is_flag=True,
+        default=False,
+        help='Any Shared Assemblies bundled into the application will be changed into Private Assemblies. '
+        'This means the exact versions of these assemblies will always be used, and any newer '
+        'versions installed on user machines at the system level will be ignored.'
+    )(cmd)
+    cmd = optgroup.option(
+        '--win-no-prefer-redirects',
+        is_flag=True,
+        default=False,
+        help='While searching for Shared or Private Assemblies to bundle into the application, '
+        'PyInstaller will prefer not to follow policies that redirect to newer versions, '
+        'and will try to bundle the exact versions of the assembly.'
+    )(cmd)
+    cmd = optgroup('Windows side-by-side assembly searching options')(cmd)
```

Comment:
How would you feel about making these two spec file only?

Reply:
That's doable. Would like to hear @rokm's opinion before doing that though.

Reply:
I think these two can be safely removed from CLI and be kept as spec-only. Actually, I'm not really sure how much of support for these assemblies is still needed these days. As I understand, old versions of VS used WinSxS for their C runtime libraries, so support for side-by-side assemblies was necessary for older python versions. But since python 3.5, VS2015+ is being used, so I don't think our SxS handling code sees much use (could be wrong, though).

Reply:
we could remove them for v5, but keep the corresponding internal code. that way if we get no complaints we can remove it in 5.2 or something, but if we do get complaints it's easy enough to re-add.

Reply:
@rokm @bwoodsend thoughts?

Reply:
Yeah, remove the CLI options but keep the backing spec arguments and the internal code.

Reply:
Sure